Posts/2021/04/tfl-WMIEventHandler.md

```diff
@@ -0,0 +1,363 @@
+---
+post_title: How Do I Discover Changes to an AD Group's Membership
+username: tfl@psp.co.uk
+Catagories: PowerShell
+tags: AD, WMI, WMI Eventing
+Summary: How to create a permanent event handler to detect changes in an AD Group
+---
+
+**Q:** Is there an easy way to detect and changes to important the membership of AD Groups?
+
+**A:**  Easy using PowerShell 7, WMI, and the CIM Cmdlets.
+
+## WMI
+Windows Management Instrumentation (WMI) is an important component of the Windows operating system.
+WMI is an infrastructure of both management data and management operations on Windows-based computers.
+You can use PowerShell to retrieve information about your host, such as the BIOS Serial number.
+Additionally, you can perform management actions, such as creating an SMB share.
+
+WMI is, in many cases, just another way to do things.
+For example, you can use WMI to create an SMB share by using the `Create` method of the **Win32_Share** class.
+For more information, see the (class documentation page for the class)[https://docs.microsoft.com/windows/win32/cimwin32prov/win32-share].
+In most cases, you use PowerShell cmdlets, such as the SMB cmdlets, to manage your SMB shares.
+The value of WMI is that it can provide you access to more information and features that are not available using cmdlets. 
+
+In writing this article, I assume you have an understanding of WMI.
+In specific, I assume you understand WMI namespaces, classes, properties, and methods.
+If not, you might like to look at the [WMI Documentation](https://docs.microsoft.com/windows/win32/wmisdk/wmi-start-page)
+And for some more details on using WMI and Powershell, look at using PowerShell 7 and WMI, look at [my recently published PowerShell 7 book](https://www.wiley.com/en-gb/PowerShell+7+for+IT+Professionals-p-9781119644705).
+In chapter 9, I devote a chapter to WMI and using the CIM cmdlets.
+To see just the scripts for that chapter, see my [GitHub repository](https://github.com/doctordns/Wiley20/tree/master/09%20-%20WMI).
+The scripts show you the basics of WMI and PowerShell 7.
```

Comment:
One of the tenets of this blog is that we don't allow marketing. I am OK with linking to your book, in the right context.  See my suggestion at the bottom of the article. I think it reads better there and comes off like a pitch for your book.

```suggestion

```